hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutAllPerMemberOperation.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation;
+
+import com.hazelcast.map.impl.MapEntries;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.OperationAccessor;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.Future;
+
+/**
+ * Inserts the {@link MapEntries} for all partitions of a member via locally invoked {@link PutAllOperation}.
+ * <p/>
+ * Used to reduce the number of remote invocations of an {@link com.hazelcast.core.IMap#putAll(Map)} call.
+ */
+public class PutAllPerMemberOperation extends MapOperation implements DataSerializable {
```

Comment:
I would make this operation IdentifiedDataSerializable. All operations are automatically DataSerializable; so it isn't needed to repeat it.